src/I18n/Parser/PoFileParser.php

```diff
@@ -142,11 +142,10 @@ protected function _addMessage(array &$messages, array $item)
 
         $translation = stripcslashes($translation);
 
-        if ($context && (!isset($messages[$singular]) || is_array($messages[$singular]))) {
+        if ($context !== null) {
             $messages[$singular]['_context'][$context] = $translation;
-        }
-        if ($context === null) {
-            $messages[$singular] = $translation;
+        } else {
+            $messages[$singular]['_context'][''] = $translation;
```

Comment:
Do you think arrays in modern PHP cannot handle this gracefully? I suspect this to be none, since the additional data is minimal.